lms/static/js/edxnotes/plugins/ai_summarize.js

```diff
@@ -0,0 +1,101 @@
+                        background-color: white;
+                        display: block;
+                        position: absolute;
+                        top: 100%;
+                        left: 5px;
+                        border-bottom: 1px solid gray;
+                        border-right: 1px solid gray;
+                        z-index: 0;
+                        transform: translateY(-50%) rotate(45deg);
+                    }
+
+                    .annotator-adder button::before,
+                    .annotator-adder button::after {
+                        display: none !important;
+                    }
+
+                    .annotator-adder #annotateButton,
+                    .annotator-adder #summarizeButton {
+                        border: none !important;
+                        background: rgb(0, 48, 87) !important;
+                        box-shadow: none !important;
+                        width: initial;
+                        transition: color .1s;
+                        text-indent: initial;
+                        font-size: 20px;
+                        padding: 0;
+                        height: fit-content;
+                        color: white;
+                        border-radius: 5px;
+                        padding-left: 5px;
+                        padding-right: 5px;
+                        font-weight: normal;
+                    }
+
+                    .annotator-adder button i.fa {
+                        font-style: normal;
+                    }
+
+                    .annotator-adder {
+                        width: fit-content;
+                        height: fit-content;
+                        padding: 5px;
+                        background-color: white;
+                        border: 1px solid gray;
+                        gap: 5px;
+                    }
+                    `;
+                document.head.appendChild(style);
+                var annotator = this.annotator;
+                annotator.adder[0].children[0].id = 'annotateButton';
+                annotator.adder[0].children[0].innerHTML = '<i class="fa fa-pencil" aria-hidden="true"></i>';
+                annotator.adder[0].innerHTML += `
+                <button class="summarize-button" id="summarizeButton" title="Summarize text using AI.">
+                    <i class="fa fa-star" aria-hidden="true"></i>
+                </button>
+                `;
+                function onDisplayStyleChange(mutationsList, observer) {
+                    for (const mutation of mutationsList) {
+                      if (mutation.type === 'attributes' && mutation.attributeName === 'style') {
+                        if (mutation.target.style.display == 'block') {
+                            mutation.target.style.display = 'flex';
+                        };
+                      };
+                    };
+                };
+                const observer = new MutationObserver(onDisplayStyleChange);
```

Comment:
Why is this mutation observer necessary? @alexjmpb

Reply:
I found a way around.